README.md

```diff
@@ -1,31 +1,170 @@
 - Go to Settings | Tools | Large Language Models and enter your API key in the "OpenAI Key" field. If you are a member
   of only one organization, leave the "OpenAI Organization" field empty
-- Select a code fragment, press Alt-Enter and select "apply custom edit" intention
-- Write an instruction for the LLM and wait for the result
+
+## ChatGPT communication configuration
+
+The communication with ChatGPT is achieved through OpenAI API. This API is not for free, and it does require a developer
+key. The configuration is very straight forward, all you need to do is to make sure you have the following environment
+variables set:
+
+```shell
+OPENAI_API_KEY
+OPENAI_ORGANIZATION
+```
+
+If you forgot to set up your environment variables, when you first try to invoke the Extract Function with ChatGPT,
+the following popup will appear:
+![openAiAuthFailed .png](readme-images/pngs/OpenAIAuthorizationFailed.png)
+
+From here, you can click on *"Configure OpenAI key"* and a new configuration window will show as follows:
+![openAiKeyConfig.png](readme-images%2Fpngs%2FopenAiKeyConfig.png)
 
 ## Triggering Extract Function with ChatGPT
-To trigger the Extract Function with ChatGPT simply right click inside a function, without selecting anything
-then, select *Show Context Action -> Extract Function Experiment*. This action will automatically select the entire
-function, including the doc string, and it will form a ChatGPT prompt. When ChatGPT's reply comes back,
-the first suggestion will be selected, and, if there is any viable suggestion, automatically the block of
-code will be selected and extracted in a new function.
-
-Additionally, ChatGPT can be also triggered with a selection, however it is better to be triggered with an 
-entire function, because this way it has more context and can provide better suggestions.
\ No newline at end of file
+
+To trigger the Extract Function with ChatGPT simply right click inside a function, select *Show Context Action ->
+Extract Function Experiment*. This action will automatically select the entire
+function, including the doc string, and it will form a ChatGPT prompt. When ChatGPT's reply comes back, a popup window
+is shown with the Extract Function candidates so that the developer can choose any of it (or none).
+
+Here's an example of the GUI showing two Extract Function candidates:
+![efKotlinExample1.png](readme-images%2Fpngs%2FefKotlinExample1.png)
+
+Note that in the GUI there are two sub-panels. In the first one there's a list of all the candidates (in the example
+abovre there are two).
+
+This list has two components:
+
+1. First is the number of lines of code that would be extracted
+2. The name of the new function
+
+Below this list, there's a panel showing the new function's signature. It includes:
+
+1. The access modifiers (private, public, etc.)
+2. The name of the function
+3. The list of parameters
+4. The return value (if any)
+5. The body of the function is just three dots (...)
```

Comment:
Could you please clarify that actual code for extraction is highlighted in the editor?